superset-frontend/src/components/Datasource/DatasourceModal.tsx

```diff
@@ -152,9 +152,7 @@ const DatasourceModal: FunctionComponent<DatasourceModalProps> = ({
             metric_name: metric.metric_name,
             metric_type: metric.metric_type,
             d3format: metric.d3format || null,
-            currency: !isDefined(metric.currency)
-              ? null
-              : JSON.stringify(metric.currency),
+            currency: !isDefined(metric.currency) ? null : metric.currency,
```

Comment:
can this just be 
```suggestion
            currency: metric.currency,
```
or is there a particular reason that we need to change undefined to null?

Reply:
@eschutho I'm actually not sure, here I'm just removing `JSON.stringify()` from the previous logic. I'll see if I can do some more manual testing around this.